From patchwork Thu Jul 31 21:36:44 2025
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Brandon Brnich <b-brnich@ti.com>
X-Patchwork-Id: 26854
X-Patchwork-Delegate: vigneshr@ti.com
From: Brandon Brnich <b-brnich@ti.com>
Subject: [tiL6.12 v1 2/2] PENDING: arm64: dts: ti: k3-j722s-main: Add E5010
 JEPG Encoder
Date: Thu, 31 Jul 2025 16:36:44 -0500
Message-ID: <20250731213644.2168308-3-b-brnich@ti.com>
In-Reply-To: <20250731213644.2168308-1-b-brnich@ti.com>
References: <20250731213644.2168308-1-b-brnich@ti.com>
Return-Path: nm@ti.com
List-Post: <mailto:linux-patch-review@list.ti.com>
MIME-Version: 1.0
To: linux-patch-review@list.ti.com
Cc: Praneeth Bajjuri <praneeth@ti.com>, Devarsh Thakkar <devarsht@ti.com>,
 Udit Kumar <u-kumar1@ti.com>, Darren Etheridge <detheridge@ti.com>
List-ID: <linux-patch-review@list.ti.com>
List-Post: <mailto:linux-patch-review@list.ti.com>

Enable IMG E5010 JPEG Encoder that supports baseline encoding for
semiplanar YUV420 and YUV422.

Signed-off-by: Brandon Brnich <b-brnich@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j722s-main.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi b/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi
index b0386913c932..3d967d3ec0f4 100644
--- a/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j722s-main.dtsi
@@ -401,6 +401,17 @@ c7x_1: dsp@7e200000 {
 		ti,sci-proc-ids = <0x31 0xff>;
 		status = "disabled";
 	};
+
+	e5010: e5010@fd20000 {
+		compatible = "img,e5010-jpeg-enc";
+		reg = <0x00 0xfd20000 0x00 0x100>,
+		      <0x00 0xfd20200 0x00 0x200>;
+		reg-names = "core", "mmu";
+		clocks = <&k3_clks 201 0>;
+		clock-names = "core_clk";
+		power-domains = <&k3_pds 201 TI_SCI_PD_EXCLUSIVE>;
+		interrupts = <GIC_SPI 98 IRQ_TYPE_LEVEL_HIGH>;
+	};
 };
 
 &main_bcdma_csi {
